lib/rss.js

```diff
@@ -0,0 +1,192 @@
+                'property:display_name': rhs.display_name,
+              },
+              {
+                'property:id': rhs.id,
+              },
+              {
+                'property:github_account': rhs.github_account,
+              },
+            ],
+          })
+        }
+
+        if (itemPath[0] === 'leaders') {
+          return
+        }
+
+        if (finalProperty === 'completed_task_instance_count') {
+          title = `Completed Tasks Updated for ${organization.name}`
+        } else if (finalProperty === 'rank') {
+          title = `Organization Rank Updated for ${organization.name}`
+        } else {
+          title = `Updated "${
+            kind === 'A' ? itemPath[itemPath.length - 2] : finalProperty
+          }" for ${organization.name}`
+        }
+
+        if (kind === 'A') {
+          return feedItems.push({
+            title,
+            date: dataUpdated,
+            custom_elements: [
+              {
+                'org:name': organization.name,
+              },
+              {
+                'org:slug': organization.slug,
+              },
+              {
+                'org:id': organization.id,
+              },
+              {
+                'property:path': stringPath,
+              },
+              {
+                'property:type': kind === 'N' ? 'addition' : 'deletion',
+              },
+              {
+                'property:item': rhs,
+              },
+            ],
+          })
+        }
+
+        return feedItems.push({
+          title,
+          description: `${
+            organization.name
+          } changed ${finalProperty} from ${lhs} to ${rhs}.`,
+          date: dataUpdated,
+          custom_elements: [
+            {
+              'org:name': organization.name,
+            },
+            {
+              'org:slug': organization.slug,
+            },
+            {
+              'org:id': organization.id,
+            },
+            {
+              'property:path': stringPath,
+            },
+            {
+              'property:type': 'change',
+            },
+            {
+              'property:old': lhs,
+            },
+            {
+              'property:new': rhs,
+            },
+          ],
+        })
+      })
+
+      const feed = new RSS({
+        title: 'Google Code-in Leaders',
+        description: 'A feed for Google Code-in updates',
+        site_url: process.env.URL,
+        feed_url: `${process.env.URL}/build.xml`,
+        pubDate: new Date(),
+        custom_namespaces: {
+          org: 'https://g.co/gci',
+          property: 'https://g.co/gci',
+        },
+      })
+
+      feedItems.forEach(item => feed.item(item))
+
+      const feedData = {
+        items: feedItems,
+        lastUpdated: dataUpdated,
+      }
+
+      fs.writeFileSync(
+        `${__dirname}/../out/feed_items.json`,
+        JSON.stringify(feedData)
+      )
+      fs.writeFileSync(`${__dirname}/../out/feed.xml`, feed.xml())
```

Comment:
if this belongs in the `main`, the test can be isolated without mocking fs calls

Reply:
I agree; separating generation from the layout of the files is a good idea.